File: Bidhouse/Bidhouse-SPA/src/viewModels/CreatePostInputModel.ts
import { Injectable } from "@angular/core";


export class CreatePostInputModel {
    
    Name:any;
    Description:any;
    Category:any;
    Location:any;
    Time:any;
    Price:any;

    constructor(name:string,description:string,category:number,location:string,time:any,price:number) {
      this.Name = name,
      this.Description = description,
      this.Category = category;
      this.Location = location,
      this.Time = time,
      this.Price = price;
    }

    
}

File: Bidhouse/ViewModels/PostModels/CreatePostInputModel.cs
using Bidhouse.Models;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace Bidhouse.ViewModels.PostModels
{
    public class CreatePostInputModel
    {
        [Required(ErrorMessage = "You should provide a post name")]
        [StringLength(70, MinimumLength = 10, ErrorMessage = "Name length must be between 10 and 70 characters")]
        public string Name { get; set; }
        [Required(ErrorMessage = "You should provide a description")]
        [StringLength(270, MinimumLength = 10, ErrorMessage = "Name length must be between 10 and 270 characters")]
        public string Description { get; set; }
        [Required(ErrorMessage ="You should provide a location")]
        public string Location { get; set; }
        [Required(ErrorMessage = "You should provide a budget ")]
        [Range(0,5000000000,ErrorMessage = "Your budget cannot be 0")]
        public decimal Price { get; set; }
        [Required(ErrorMessage = "You should choose a category")]
        public Category Category { get; set; }
        [Required(ErrorMessage = "You should choose a date of delivery")]
        public DateTime Time { get; set; }
    }
}
